### PR TITLE
[#165] Add tests for local references

### LINKED ISSUE #165: Test all variants of local references
# Clarification and motivation

Let's extend the bats tests a bit to test some basic local references:
* "self links" to an anchor within the same file: `#anchor`
* "relative links" with and without anchors: `dir/file.md`, `dir/file.md#anchor`, `./dir/file.md`, `dir/./file.md`, `../dir/file.md`
* "absolute links" with and without anchors: `/dir/file.md`, `dir/file.md#anchor`, `/dir/./file.md`
* Test all of this when the root directory matches the current directory (i.e. the default `--root .`) **and** when it doesn't (i.e. `--root somedir`)

The tests in `tests/golden/check-anchors` already cover the specific case of relative links with anchors when `--root .`, but I think those tests should be replaced with a more complete suite for _all_ kinds of local references

# Acceptance criteria

* We have bats tests for all variants of local references

### PR DESCRIPTION
**Pull request by Sorokin-Anton**
## Description
Problem: We want all cases of local references to be covered by bats tests (e.g. cause we want to be sure that all this works well on Windows), but there was only one case covered in `check-anchors`

Solution: make a `check-local-refs` folder and test different types of links  with different roots

## Related issue(s)

Fixes #165 

## :white_check_mark: Checklist for your Pull Request

Ideally a PR has all of the checkmarks set.

If something in this list is irrelevant to your PR, you should still set this
checkmark indicating that you are sure it is dealt with (be that by irrelevance).

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](https://github.com/serokell/xrefcheck/tree/master/README.md)
    - Haddock

- Public contracts
  - [x] Any modifications of public contracts comply with the [Evolution
  of Public Contracts](https://www.notion.so/serokell/Evolution-of-Public-Contracts-2a3bf7971abe4806a24f63c84e7076c5) policy.
  - [x] I added an entry to the [changelog](https://github.com/serokell/xrefcheck/tree/master/CHANGES.md) if my changes are visible to the users
        and
  - [x] provided a migration guide for breaking changes if possible

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [x] My code complies with the [style guide](https://github.com/serokell/style/blob/master/haskell.md).

#### ✓ Release Checklist

- [ ] I updated the version number in `package.yaml`.
- [ ] I updated the [changelog](https://github.com/serokell/xrefcheck/tree/master/CHANGES.md) and moved everything
      under the "Unreleased" section to a new section for this release version.
- [ ] (After merging) I edited the [auto-release](https://github.com/serokell/xrefcheck/releases/tag/auto-release).
    * Change the tag and title using the format `vX.Y.Z`.
    * Write a summary of all user-facing changes.
    * Deselect the "This is a pre-release" checkbox at the bottom.
- [ ] (After merging) I updated [`xrefcheck-action`](https://github.com/serokell/xrefcheck-action#updating-supported-versions).
- [ ] (After merging) I uploaded the package to [hackage](https://hackage.haskell.org/package/xrefcheck).

### CODE REVIEW

**Comment by Sorokin-Anton** [comment]
@dcastro  during rebases I accidently marked PR #186 as merged (without actual merge), so I am creating same new one 😐

**Comment by dcastro** [comment]
> @dcastro during rebases I accidently marked PR #186 as merged (without actual merge), so I am creating same new one neutral_face

Ah I see :sweat_smile: 

When I have a PR X piggybacking on top of another PR Y, like you have here, I usually do this:
* Merge X into master first
* Rebase Y on top of master
* Merge Y into master

File: tests/golden/check-local-refs/dir1/dir2/d2f1.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+[bad-file-rel](../a/b/c/unexisting-file.md)
+
+[bad-anchor-rel-1](d2f2.md#bad-anchor)
+[bad-anchor-rel-2](unexisting-file.md#bad-anchor)
+
+# Absolute links
+Should be correct when root is `/tests/golden/check-local-refs`:
+[file-abs-1](/dir1/./d1f1.md)
+[anchor-abs-1](/dir1/.././d1f1.md#existing-anchor-d1f1)
```
Comment (dcastro) [inline tests/golden/check-local-refs/dir1/dir2/d2f1.md:29]: This link isn't valid (and it does indeed appear as an invalid link in `expected1.gold`), but in line 27 we say that it "Should be correct".

Comment (Sorokin-Anton) [reply tests/golden/check-local-refs/dir1/dir2/d2f1.md]: My bad, thanks, fixed

**Pull request merged**